[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56580 [JIT] Enable conv-add-relu fusion as a part of frozen graph optimization**
* #56579 [JIT] Disable conv-add-relu fusion for cuDNN7 when model uses fp16

Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization.
Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

desertfire committed:
[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, still controlled by the optimize_numerics option

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

eellison reviewed (approved):
LGTM. Technically, we're burning in build settings here into the freezing, so this would be a good thing to move to the `mach=native` setting or whatever in the freezing API since we are burning in cudnn being available but I dont want to block this being added.

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization, but it is still controlled by the optimize_numerics option.

Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization.
Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]

desertfire committed:
Update on "[JIT] Enable conv-add-relu fusion as a part of frozen graph optimization"


Summary: Turn on conv-add-relu fusion as default for the frozen graph optimization.
Test Plan:
```
python test/test_jit.py -k test_freeze_conv_relu_fusion
```

Differential Revision: [D27915515](https://our.internmc.facebook.com/intern/diff/D27915515)

[ghstack-poisoned]
